qa/L0_libtorch_shared_weights/test.sh

```diff
@@ -0,0 +1,204 @@
+# Copyright 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REPO_VERSION=${NVIDIA_TRITON_SERVER_VERSION}
+if [ "$#" -ge 1 ]; then
+    REPO_VERSION=$1
+fi
+if [ -z "$REPO_VERSION" ]; then
+    echo -e "Repository version must be specified"
+    echo -e "\n***\n*** Test Failed\n***"
+    exit 1
+fi
+if [ ! -z "$TEST_REPO_ARCH" ]; then
+    REPO_VERSION=${REPO_VERSION}_${TEST_REPO_ARCH}
+fi
+
+export CUDA_VISIBLE_DEVICES=0
+
+CLIENT_LOG="./client.log"
+DATADIR=/data/inferenceserver/${REPO_VERSION}
+INSTANCE_CNT=16
+REUSE_MSG="Reusing TorchScript model for instance"
+SERVER=/opt/tritonserver/bin/tritonserver
+SERVER_ARGS="--model-repository=`pwd`/models --exit-on-error=false \
+             --exit-timeout-secs=10"
+TEST_RESULT_FILE='test_results.txt'
+WEIGHTS_TEST=libtorch_shared_weights_test.py
+source ../common/util.sh
+
+RET=0
+rm -fr *.log
+
+LOG_IDX=0
+
+# Prepare model repository
+rm -fr models
+mkdir models
+for i in models; do
+    cp -r $DATADIR//qa_identity_model_repository/libtorch_nobatch_zero_1_float32 models/.
+done
+
+for MC in `ls models/libtorch*/config.pbtxt`; do
+    echo "instance_group [ { count: ${INSTANCE_CNT} kind: KIND_GPU}]" >> $MC
+done
+
+# SharedWeightsTest.test_pytorch_identity_model
```

Comment:
Instead of duplicating the code for gpu and cpu, you can run it in a loop. 
See how we do it in L0_infer:
https://github.com/triton-inference-server/server/blob/main/qa/L0_infer/test.sh#L132
https://github.com/triton-inference-server/server/blob/main/qa/L0_infer/test.sh#L238

Reply:
Thanks for the tip! Updated.